ubyssey/views/guide.py

```diff
@@ -1,22 +1,137 @@
 from django.shortcuts import render
 from django.http import Http404
-
+import json
+from django.utils.safestring import SafeString
+from dispatch.models import Article, Tag
+from django.urls import reverse
 import ubyssey
 from ubyssey.helpers import ArticleHelper
 
-class GuideTheme(object):
+class Guide2016(object):
     """Theme for the 2016 Ubyssey Guide to UBC."""
 
-    def __init__(self, year, title): 
+    def __init__(self, title): 
+        self.title = title
+
+    def landing(self, request):
+        """The Guide to UBC landing page."""
+        landing_page = 'guide/2016/index.html'
+        return render(request, landing_page, {})
+
+    def article(self, request, slug=None):
+        """Guide article page."""
+        try:
+            article = ArticleHelper.get_article(request, slug)
+        except:
+            raise Http404('Article could not be found.')
+
+        template_fields = article.template_fields
+
+        try:
+            next_a = ArticleHelper.get_article(request, template_fields['next_a'])
+        except:
+            next_a = None
+
+        try:
+            next_b = ArticleHelper.get_article(request, template_fields['next_b'])
+        except:
+            next_b = None
+
+        article.add_view()
+
+        context = {
+            'title': article.headline,
+            'meta': ArticleHelper.get_meta(article),
+            'article': article,
+            'next': [next_a, next_b]
+        }
+        article_page = 'guide/2016/article.html'
+        return render(request, article_page , context)
+
+
+class Guide2020(object):
+    """Theme for the 2016 Ubyssey Guide to UBC."""
+
+    def __init__(self, year, title, section1_name, section2_name, section3_name, section4_name, section5_name): 
         self.year = str(year)
         self.title = title
+        self.section1_name = section1_name
+        self.section2_name = section2_name
+        self.section3_name = section3_name
+        self.section4_name = section4_name
+        self.section5_name = section5_name
```

Comment:
Could this be a list rather than separate variables?

```py
self.sections = sections
```

Reply:
I think it ought to be (unless there's good reason for it not to be that I am not seeing)